### ISSUE TITLE
USB 2.0 Audio Stream Isochronous packet loss

### ISSUE THREAD

**Issue by sreimers** [raspberrypi/linux#326]
Hi together,

https://gist.github.com/sreimers/5942827#file-usb_audio_defect-txt

```
caf23460 1574370683 C Zo:1:004:2 0:1:0:0 5 0:0:1 0:176:180 0:352:180 0:528:180 0:704:180 884 >
caf23460 1574370749 S Zo:1:004:2 -115:1:0 8 -18:0:176 -18:176:176 -18:352:176 -18:528:180 -18:708:176 1412 = f5fb00ff aafb51ff 61fb94ff 32fbcaff 1ffbf1ff 2efbf6ff 5bfbe4ff a4fbc5ff
```

The "Completion" and the "Submit" length differs from 8 to 5 bytes often. This results in constant audio pops and clicks. On other USB Hosts (e.g. my laptop etc.) I see a constant 8 Byte length.

https://gist.github.com/sreimers/5942827#file-usb_audio_good-txt

More debug information you can find here: https://gist.github.com/sreimers/5942827

For testing I used, the current FIQ, Next and Master Firmware Branch.

The actually debug information on this ticket generated with this kernel:

```
Linux podlove 3.9.8+ #490 PREEMPT Tue Jul 2 18:07:10 BST 2013 armv6l GNU/Linux
```

If I reduce the audio frequency to 32 kHz, capture and playback are in parallel OK. Also if I use 44,1 kHz only for playback. But playback 44,1 kHz and capture 44,1kHz is a no go ;-)

**Comment by P33M** [comment]
This is a known limitation with fiq_split

See https://github.com/raspberrypi/linux/issues/351

**Comment by P33M** [comment]
If you add dwc_otg.fiq_split_enable=0 to the /boot/cmdline.txt parameters, you should be able to semi-reliably capture at the rates of interest. There will be occasional data loss due to interrupt latency however.

Closing as duplicate of (known) issue.
